enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
+        """
+        existing_users, unregistered_emails = cls.get_users_by_email(emails)
+        course_ids = get_course_runs_from_program(program_details)
+
+        successes = []
+        pending = []
+        failures = []
+
+        for user in existing_users:
+            succeeded = cls.enroll_user(enterprise_customer, user, course_mode, *course_ids)
+            if succeeded:
+                successes.append(user)
+            else:
+                failures.append(user)
+
+        for email in unregistered_emails:
+            pending_user = cls.enroll_user_pending_registration(enterprise_customer, email, course_mode, *course_ids)
+            pending.append(pending_user)
+
+        return successes, pending, failures
+
+    @classmethod
+    def enroll_users_in_course(cls, enterprise_customer, course_id, course_mode, emails):
+        """
+        Enroll existing users in a course, and create a pending enrollment for nonexisting users.
+
+        Args:
+            enterprise_customer: The EnterpriseCustomer which is sponsoring the enrollment
+            course_id (str): The unique identifier of the course in which we're enrolling
+            course_mode (str): The mode with which we're enrolling in the course
+            emails: An iterable of email addresses which need to be enrolled
+
+        Returns:
+            successes: A list of users who were successfully enrolled in the course
+            pending: A list of PendingEnterpriseCustomerUsers who were successfully linked and had
+                pending enrollments created for them in the database
+            failures: A list of users who could not be enrolled in the course
+        """
+        existing_users, unregistered_emails = cls.get_users_by_email(emails)
+
+        successes = []
+        pending = []
+        failures = []
+
+        for user in existing_users:
+            succeeded = cls.enroll_user(enterprise_customer, user, course_mode, course_id)
+            if succeeded:
+                successes.append(user)
+            else:
+                failures.append(user)
+
+        for email in unregistered_emails:
+            pending_user = cls.enroll_user_pending_registration(enterprise_customer, email, course_mode, course_id)
+            pending.append(pending_user)
+
+        return successes, pending, failures
+
+    @classmethod
+    def send_messages(cls, http_request, message_requests):
+        """
+        Deduplicate any outgoing message requests, and send the remainder.
+
+        Args:
+            http_request: The HTTP request in whose response we want to embed the messages
+            message_requests: A list of undeduplicated messages in the form of tuples of message type
+                and text- for example, ('error', 'Something went wrong')
+        """
+        deduplicated_messages = set(message_requests)
+        for msg_type, text in deduplicated_messages:
+            getattr(messages, msg_type)(http_request, text)
```

Comment:
I see what is happening here, but it would be nice to break up a bit for readability.  Can we set the result of `getattr()` to a variable and then call `variable(http_request, text)` ?